packages/docs/docs/hosting/heroku.mdx

```diff
@@ -0,0 +1,84 @@
+# Heroku
+
+## Overview
+
+For EzBackend to be compatible with hosting on Heroku, there needs to be some additional configuration.
+
+The below configuration is for the recommended configuration on Heroku, `EzBackend` + `Postgres`. You will probably need to configure all the items below for EzBackend to work as expected
+
+## Port Binding
+
+### Fix
+
+In `app.start`
+
+```ts
+app.start({
+  backend: {
+    listen: {
+      address: '0.0.0.0',
+    },
+  },
+});
+```
+
+### Explanation
+
+By default, EzBackend will only listen on address `127.0.0.1`. Since Heroku assigns a random-ish port to each instance, it will throw the error
+
+```
+Error R10 (Boot timeout) -> Web process failed to bind to $PORT within 60 seconds of launch
+```
+
+To fix this, we need to listen on all addresses, or address `0.0.0.0`
+
+## Auth Callback Route
+
+### Fix
+
+In `.env`
+
+```bash
+PRODUCTION_URL=<Your heroku endpoint e.g. https:my-app.herokuapp.com>
+```
+
+### Explanation
+
+By default, EzBackend tries to obtain your url in order to properly set the callback URL for authentication callback routes. However, this results in EzBackend detecting the url without `https`
+
+We can however specify the `PRODUCTION_URL` manually in order to override the automated URL detection, which will allow authentication to properly work
+
+## Database Persistence
+
+### Fix
+
+In `Heroku`
+
+1. Create an instance of Heroku postgres
+2. Share the instance of Heroku postgres with your EzBackend deployment
+3. Share the URL of the Heroku postgres instance as an environment variable `DATABASE_URL`
+4. Turn on SSL for Heroku postgres
+
+In `app.start`
+
+```ts
+app.start({
+  backend: {
+    typeorm: {
+      type: 'postgres',
+      url: process.env.DATABASE_URL,
+      ssl: true,
+    },
+  },
+});
+```
+
+### Explanation
+
+Heroku's may spin up and down your instances at any time, and EzBackend by default stores data in SQLite, with files stored in the instance itself.
+
+While this is good for getting started quickly, it is not optimal for production workloads because you will lose your data whenever an instance is restarted and your data is lost.
+
+Hence, it is best to create a `heroku postgres` instance and connect your database to there instead
```

Comment:
maybe can also add an alternative step: attach db instance using heroku web? i imagine that there may be some users who are not familiar with heroku CLI so they may use web instead

Reply:
Yeah actually adding the postgres addon automatically added/changed the `DATABASE_URL` to the correct one